src/URI/ByteString/Types.hs

```diff
@@ -52,9 +52,24 @@ data UserInfo = UserInfo {
 
 
 -------------------------------------------------------------------------------
-newtype Query = Query { queryPairs :: [(ByteString, ByteString)] }
-              deriving (Show, Eq, Monoid, Generic, Typeable, Ord)
-
+data Query
+    = Query { queryPairs :: [(ByteString, ByteString)] }
```

Comment:
Mixing sum types with record accessors is generally frowned upon as it produces partial functions that throw at runtime. We may want to look into removing the record notation and perhaps expose Prism'like combinators to get at the arguments.